tests/application/cms/models/test_data_source_model.py

```diff
@@ -87,3 +89,53 @@ def test_search_should_be_ordered_by_number_of_associated_measure_versions(self)
         assert (
             data_source_associated_with_1_measure_version == search[1]
         ), "Expected data source associated with 1 measure version to be second"
+
+    def test_merge_from_single_other_data_source(self):
+
+        data_source_1 = DataSourceFactory.create()
+        data_source_2 = DataSourceFactory.create()
+
+        measure_version_associated_with_data_source_2 = MeasureVersionFactory.create(data_sources=[data_source_2])
+
+        assert data_source_1.merge(data_source_ids=[data_source_2.id]) is True
+        assert DataSource.query.get(data_source_2.id) is None
+
+        assert measure_version_associated_with_data_source_2.data_sources == [data_source_1]
+
+    def test_merge_from_two_other_data_sources(self):
+
+        data_source_1 = DataSourceFactory.create()
+        data_source_2 = DataSourceFactory.create()
+        data_source_3 = DataSourceFactory.create()
+
+        measure_version_associated_with_data_source_2 = MeasureVersionFactory.create(data_sources=[data_source_2])
+
+        measure_version_associated_with_data_source_3 = MeasureVersionFactory.create(data_sources=[data_source_2])
+
+        assert data_source_1.merge(data_source_ids=[data_source_2.id, data_source_3.id]) is True
+        assert DataSource.query.get(data_source_2.id) is None
+        assert DataSource.query.get(data_source_3.id) is None
+
+        assert measure_version_associated_with_data_source_2.data_sources == [data_source_1]
+        assert measure_version_associated_with_data_source_3.data_sources == [data_source_1]
```

Comment:
This test feels a bit weird - I don't see any difference between data source/measure version 2 and 3? I assumed that they'd each be testing a slightly different thing, but I can't see why we need the third one really. Might be simpler to drop it? Unless I'm missing something obvious.

Reply:
There's no difference - it's just testing that you can merge two different data sources at once. Possibly overkill?